mmv1/third_party/terraform/tpgresource/common_diff_suppress.go.erb

```diff
@@ -179,25 +179,47 @@ func TimestampDiffSuppress(format string) schema.SchemaDiffSuppressFunc {
 	}
 }
 
-// suppress diff when saved is Ipv4 and Ipv6 format while new is required a reference
-// this happens for an internal ip for Private Services Connect
+// Suppresses diff for IPv4 and IPv6 different formats.
+// It also suppress diffs if one IP and one reference are given.
 func InternalIpDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
-	olds := strings.Split(old, "/")
-	news := strings.Split(new, "/")
-
-	if len(olds) == 2 {
-		if len(news) == 2 {
-			return bytes.Equal(net.ParseIP(olds[0]), net.ParseIP(news[0])) && olds[1] == news[1]
-		} else {
-			return (net.ParseIP(olds[0]) != nil) && (net.ParseIP(new) == nil)
+	addr_netmask_old := strings.Split(old, "/")
+	addr_netmask_new := strings.Split(new, "/")
+
+	// Check if old or new are IPs (with or without netmask)
+	addr_old := func() net.IP {
```

Comment:
Why are these being handled in closures? That's not a common pattern in this codebase.

Reply:
done